### PR TITLE
feat: [SRM-15340]: Added Report Event Card

### PR DESCRIPTION
**Pull request by deepesh-ui**
#### Screenshots

Screenshot 2023-08-23 at 10 43 45 AM

#### PR Checklist

<details>
<summary>Please check if your PR fulfils the following requirements</summary>

- [ ] Tests for the changes have been added. Ideally, include a test that fails without this PR but passes with it.
- [ ] Docs have been [added/updated](https://harness.atlassian.net/jira/software/c/projects/DOC/boards/40).
</details>

<details>
<summary>Use the following comments to re-trigger PR Checks</summary>

- Jest: `retrigger jest`
- Prettier: `retrigger prettier`
- Type Check: `retrigger typecheck`
- ESLint: `retrigger eslint`
- Sonar: `retrigger sonar`
- Standards: `retrigger standards`
- Build: `retrigger build`
- Title Check: `retrigger titlecheck`
- Feature Name Check: `trigger featurenamecheck`
- Coverage: `retrigger coverage`
- Rebase: `trigger rebase`
- Cypress Rest: `retrigger cypress-rest`
- Cypress CD: `retrigger cypress-cd`
- Cypress Pipeline: `retrigger cypress-pipeline`
- Cypress CV: `retrigger cypress-cv`
- Fix Prettier: `fix prettier`
</details>

### CODE REVIEW

File: src/modules/85-cv/pages/monitored-service/components/ServiceHealth/components/ReportsTable/ReportDrawer/Components/ReportDetails/ReportDetails.tsx [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+        margin={{ right: 'medium' }}
+        color={Color.BLACK_100}
+      >
+        {getString('details')}
+      </Text>
+
+      <Container className={css.gridContainer}>
+        <Container className={css.gridLayout}>
+          <Text font={{ size: 'small' }}>{getString('common.executedBy')}</Text>
+          <ExecutedBy planExecutionId={'e1cj-BXUSteR4u0o6rZlpQ'} stageNodeId={'gqg2mnAjRJOBDLMrLOnjFA'} />
```
Comment (deepesh-ui) [inline src/modules/85-cv/pages/monitored-service/components/ServiceHealth/components/ReportsTable/ReportDrawer/Components/ReportDetails/ReportDetails.tsx:46]: Waiting for a BE to Send planExecutionId and planExecutionId

File: src/modules/85-cv/pages/monitored-service/components/ServiceHealth/components/ReportsTable/ReportDrawer/Components/ReportDetails/Components/ExecutedBy.tsx [hunk 2]
```diff
@@ -0,0 +1,86 @@
...
+      <Text font={{ size: 'small' }} margin={{ left: 'small', right: 'small' }} color={Color.ERROR}>
+        {getErrorMessage(executionError)}
+      </Text>
+    )
+  } else {
+    content = (
+      <>
+        <UserLabel
+          name={identifier || extraInfo?.email}
+          email={extraInfo?.email}
```
Comment (harshilbits) [inline src/modules/85-cv/pages/monitored-service/components/ServiceHealth/components/ReportsTable/ReportDrawer/Components/ReportDetails/Components/ExecutedBy.tsx:61]: `extraInfo?.email` is getting repeated . Please move it to a variable.

File: src/modules/85-cv/pages/monitored-service/components/ServiceHealth/components/ReportsTable/ReportDrawer/Components/ReportSLIAndHealthSource/ReportSLIAndHealthSource.tsx [hunk 3]
```diff
@@ -0,0 +1,59 @@
...
+
+  const [timeStamps, setTimestamps] = useState<[number, number]>([0, 0])
+
+  const derivedEndTime = useMemo(() => calculateEndtime(analysisEndTime || 0), [analysisEndTime])
+
+  useEffect(() => {
+    if (analysisStartTime) {
+      setTimestamps([analysisStartTime - TWO_HOURS_IN_MILLISECONDS, derivedEndTime])
+    }
+  }, [derivedEndTime])
```
Comment (harshilbits) [inline src/modules/85-cv/pages/monitored-service/components/ServiceHealth/components/ReportsTable/ReportDrawer/Components/ReportSLIAndHealthSource/ReportSLIAndHealthSource.tsx:28]: shouldnt we have `analysisStartTime` also in dependency array ?

File: src/modules/85-cv/pages/monitored-service/components/ServiceHealth/components/ReportsTable/ReportDrawer/ReportDrawer.tsx [hunk 4]
```diff
@@ -0,0 +1,63 @@
...
+        status={analysisStatus}
+        isLoading={analysisLoading}
+      />
+      <Container margin={{ top: 'medium', bottom: 'medium' }} height={1} background={Color.GREY_200} />
+      {analysisError ? (
+        <PageError message={getErrorMessage(analysisError)} />
+      ) : (
+        <>
+          <ReportDetails
+            {...(analysisSummary?.resource as SRMAnalysisStepDetailDTO)}
```
Comment (harshilbits) [inline src/modules/85-cv/pages/monitored-service/components/ServiceHealth/components/ReportsTable/ReportDrawer/ReportDrawer.tsx:53]: `analysisSummary?.resource as SRMAnalysisStepDetailDTO` is getting repeated. Please move it to a variable.

**Review by praneshg239 — APPROVED** [review]

**Review by harshilbits — APPROVED** [review]

**Pull request merged**